monkey/common/agent_plugins/agent_plugin_repository_index.py

```diff
@@ -67,6 +63,7 @@ def _infection_monkey_version_parser(
 
         raise TypeError(f'Expected "{DEVELOPMENT}" or a valid semantic version, got {type(value)}')
 
-    @validator("plugins")
+    @field_validator("plugins")
+    @classmethod
     def _convert_str_type_to_enum(cls, plugins):
         return {AgentPluginType(t): plugins for t, plugins in plugins.items()}
```

Comment:
Do we still need this?

Reply:
Yep, we do. It will not be converted to enum if it is not there.